extensions/rapidcap/src/create-card.tsx

```diff
@@ -0,0 +1,68 @@
+import { Form, Action, ActionPanel, useNavigation, getPreferenceValues, showToast, Toast } from "@raycast/api";
+import { FormValidation, useForm } from "@raycast/utils";
+import { useEffect, useState } from "react";
+import { Preferences, Card } from "./types";
+import fs from "fs";
+import validateTag from "./utils";
+
+function CreateCardAction() {
+  const preferences = getPreferenceValues<Preferences>();
+  const { pop } = useNavigation();
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    if (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Something went wrong",
+        message: error.message,
+      });
+    }
+  }, [error]);
+
+  const { handleSubmit, itemProps } = useForm<Card>({
+    onSubmit(values: { question: string; answer: string; tag: string }) {
+      fs.readFile(preferences.dataFile, "utf-8", (error, data) => {
+        if (error) {
+          setError(error);
+        }
+
+        const cards: Card[] = JSON.parse(data);
+        cards.push(values);
+
+        fs.writeFile(preferences.dataFile, JSON.stringify(cards, null, 4), (error) => {
+          if (error) {
+            setError(error);
+          }
+        });
+      });
+      pop();
+    },
+    validation: {
+      question: FormValidation.Required,
+      answer: FormValidation.Required,
+      tag: (value) => {
+        if (value && validateTag(value)) {
+          return "Only 1 tag is supported & it cannot contain a space!";
```

Comment:
Is there a reason why only one tag is supported? If you want, I think it could support arbitrary number of tags, separated by whitespace

Reply:
I went back and forth on this one. I am really trying to shoot for the MVP of this right now. I definitely plan to add that and quick actions (such as creating a card from the initial prompt instead of having to go to the form). But I have heard the code approval could take some time so I figured I could work on those things while I wait.